Makefile

```diff
@@ -381,34 +389,69 @@ ifeq ($(EXPORT_RESULT), true)
 
-.PHONY: test-clusterctl-delete
+.PHONY: test-cluster-delete
 test-clusterctl-delete: ## Delete clusterctl created cluster
-	kubectl -n ${TEST_NAMESPACE} delete cluster ${TEST_CLUSTER_NAME}
+	kubectl -n ${TEST_NAMESPACE} delete cluster ${TEST_CLUSTER_NAME} --ignore-not-found
 
-.PHONY: test-kubectl-bootstrap
-test-kubectl-bootstrap: ## Run kubectl queries to get all capx management/bootstrap related objects
+.PHONY: list-bootstrap-resources
+list-bootstrap-resources: ## Run kubectl queries to get all capx management/bootstrap related objects
 	kubectl get ns
 	kubectl get all --all-namespaces
 	kubectl -n capx-system get all
 	kubectl -n $(TEST_NAMESPACE) get Cluster,NutanixCluster,Machine,NutanixMachine,KubeAdmControlPlane,MachineHealthCheck,nodes
 	kubectl -n capx-system get pod
 
-.PHONY: test-kubectl-workload
-test-kubectl-workload: ## Run kubectl queries to get all capx workload related objects
+.PHONY: list-workload-resources
+list-workload-resources: ## Run kubectl queries to get all capx workload related objects
 	kubectl -n $(TEST_NAMESPACE) get secret
 	kubectl -n ${TEST_NAMESPACE} get secret ${TEST_CLUSTER_NAME}-kubeconfig -o json | jq -r .data.value | base64 --decode > ${TEST_CLUSTER_NAME}.workload.kubeconfig
 	kubectl --kubeconfig ./${TEST_CLUSTER_NAME}.workload.kubeconfig get nodes,ns
 
+.PHONY: test-cc-cluster-create
+test-cc-cluster-create: cluster-templates
+	clusterctl generate cluster ${TEST_CLUSTER_CLASS_NAME} --from ./templates/cluster-template-clusterclass.yaml -n $(TEST_NAMESPACE) > ${TEST_CLUSTER_CLASS_NAME}.yaml
+	clusterctl generate cluster ${TEST_TOPOLOGY_CLUSTER_NAME} --from ./templates/cluster-template-topology.yaml -n $(TEST_NAMESPACE) > ${TEST_TOPOLOGY_CLUSTER_NAME}.yaml
+	kubectl create ns $(TEST_NAMESPACE) --dry-run=client -oyaml | kubectl apply --server-side -f -
+	kubectl apply --server-side -f ./${TEST_CLUSTER_CLASS_NAME}.yaml
+	kubectl apply --server-side -f ./${TEST_TOPOLOGY_CLUSTER_NAME}.yaml
+
+.PHONY: test-cc-cluster-delete
+test-cc-cluster-delete:
+	kubectl -n $(TEST_NAMESPACE) delete cluster ${TEST_TOPOLOGY_CLUSTER_NAME} --ignore-not-found
+	kubectl -n $(TEST_NAMESPACE) delete secret ${TEST_TOPOLOGY_CLUSTER_NAME} --ignore-not-found
+	kubectl -n $(TEST_NAMESPACE) delete cm ${TEST_TOPOLOGY_CLUSTER_NAME}-pc-trusted-ca-bundle --ignore-not-found
+	rm ${TEST_TOPOLOGY_CLUSTER_NAME}.yaml || true
+	rm ${TEST_CLUSTER_CLASS_NAME}.yaml || true
+
+.PHONY: generate-cc-cluster-kubeconfig
+generate-cc-cluster-kubeconfig:
+	kubectl -n ${TEST_NAMESPACE} get secret ${TEST_TOPOLOGY_CLUSTER_NAME}-kubeconfig -o json | jq -r .data.value | base64 --decode > ${TEST_TOPOLOGY_CLUSTER_NAME}.workload.kubeconfig
+
+.PHONY: test-cc-cluster-install-cni
+test-cc-cluster-install-cni: generate-cc-cluster-kubeconfig
+	kubectl --kubeconfig ./${TEST_TOPOLOGY_CLUSTER_NAME}.workload.kubeconfig apply -f https://raw.githubusercontent.com/nutanix-cloud-native/cluster-api-provider-nutanix/main/test/e2e/data/cni/calico/calico.yaml
```

Comment:
Is this target being used anywhere?

Reply:
dev only usage. as discussed in the meeting, e2e uses nutanix.yaml so this file has only things for dev related tasks which we plan to move to a separate makefile or devbox later.